lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,298 @@
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        if not region:
+            module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
+        logs = boto3_conn(module, conn_type='client', resource='logs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.ProfileNotFound as e:
+        module.fail_json(msg=str(e))
+
+    state = module.params.get('state')
+
+    if state == 'present':
+
+        desc_log_group = describe_log_group(client=logs, log_group_name=module.params['log_group_name'], module=module)
+
+        return_create_log_group = {}
+        if 'logGroups' in desc_log_group:
+            for i in desc_log_group['logGroups']:
+                if module.params['log_group_name'] == i['logGroupName'] and module.params['overwrite'] is True:
+                    delete_log_group(client=logs, log_group_name=module.params['log_group_name'], module=module)
+                    return_create_log_group = create_log_group(client=logs,
+                                                               log_group_name=module.params['log_group_name'],
+                                                               kms_key_id=module.params['kms_key_id'],
+                                                               tags=module.params['tags'],
+                                                               retention=module.params['retention'],
+                                                               module=module)
+                elif module.params['log_group_name'] == i['logGroupName']:
+                    return_create_log_group = i
+
+        if not return_create_log_group:
+                    return_create_log_group = create_log_group(client=logs,
+                                                               log_group_name=module.params['log_group_name'],
+                                                               kms_key_id=module.params['kms_key_id'],
+                                                               tags=module.params['tags'],
+                                                               retention=module.params['retention'],
+                                                               module=module)
+
+        desc_log_group_snake = {}
+        for k, v in return_create_log_group.items():
+            desc_log_group_snake[_camel_to_snake(k)] = v
+
+        desc_log_group_result = dict(changed=False, stdout=desc_log_group_snake)
+        module.exit_json(**desc_log_group_result)
+
+    elif state == 'absent':
+        delete_log_group(client=logs,
+                         log_group_name=module.params['log_group_name'],
+                         module=module)
+    elif state == 'get':
```

Comment:
This logic should really go in a cloudwatchlogs_log_group_facts.py module. You can move any of the shared code - maybe describe_log_group() - into module_utils/aws/.